packages/common/src/tools/LogThrottler.ts

```diff
@@ -0,0 +1,81 @@
+import { Logger, LogLevel } from './Logger'
+
+interface LogInfo {
+  count: number
+  throttleCount: number
+  isThrottling: boolean
+}
+
+export interface LogThrottlerOptions {
+  threshold: number
+  thresholdTime: number
+  throttleTime: number
+}
+
+export class LogThrottler {
+  private readonly DEFAULT_LOG_INFO = {
+    count: 0,
+    isThrottling: false,
+    throttleCount: 0,
+  }
+
+  private readonly recentLogs: Map<string, LogInfo>
+  private readonly logger: Logger = new Logger({
```

Comment:
We should create this logger based on logger options (log level, log style). Just pass it in constructor.